pkg/controllers/management/eksupstreamrefresh/eks_upstream_refresh.go

```diff
@@ -151,7 +153,16 @@ func (e *eksRefreshController) refreshClusterUpstreamSpec(cluster *mgmtv3.Cluste
 }
 
 func GetComparableUpstreamSpec(secretsCache wranglerv1.SecretCache, cluster *mgmtv3.Cluster) (*v1.EKSClusterConfigSpec, error) {
-	_, eksService, err := controller.StartAWSSessions(secretsCache, *cluster.Spec.EKSConfig)
+	switch cluster.Status.Driver {
+	case apimgmtv3.ClusterDriverEKS:
+		return BuildEKSUpstreamSpec(secretsCache, cluster)
+	default:
+		return nil, fmt.Errorf("unsupported cloud driver")
+	}
+}
+
+func BuildEKSUpstreamSpec(secretsCache wranglerv1.SecretCache, cluster *mgmtv3.Cluster) (*v1.EKSClusterConfigSpec, error) {
```

Comment:
Don't need to export function
```suggestion
func buildEKSUpstreamSpec(secretsCache wranglerv1.SecretCache, cluster *mgmtv3.Cluster) (*v1.EKSClusterConfigSpec, error) {
```